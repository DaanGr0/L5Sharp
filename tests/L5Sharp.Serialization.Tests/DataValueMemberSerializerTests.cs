using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using L5Sharp.Types;
using NUnit.Framework;

namespace L5Sharp.Serialization.Tests
{
    [TestFixture]
    public class DataValueMemberSerializerTests
    {
        [Test]
        public void Serialize_WhenCalled_ShouldNotBeNull()
        {
            var component = Member.Create<Dint>("Test");
            var serializer = new DataValueMemberSerializer();

            var xml = serializer.Serialize(component);

            xml.Should().NotBeNull();
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void Serialize_Bool_ShouldBeApproved()
        {
            var component = Member.Create<Dint>("Test");
            var serializer = new DataValueMemberSerializer();
            
            var xml = serializer.Serialize(component);

            Approvals.VerifyXml(xml.ToString());
        }
    }
}