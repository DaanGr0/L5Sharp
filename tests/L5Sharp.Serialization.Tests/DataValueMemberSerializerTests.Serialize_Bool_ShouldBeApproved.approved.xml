<DataValueMember Name="Test" DataType="DINT" Radix="Decimal" Value="0" />